packages/cosmos-tigergraph/src/index.ts

```diff
@@ -0,0 +1,185 @@
+      method: 'POST',
+      body: `{"graph": "${this.graphname}"}`,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.username}:${this.password}`)}`,
+      },
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error! status: ${response.status}`)
+      }
+
+      return response.json()
+    }).then(data => {
+      this.token = data.results.token
+      return this.token
+    })
+  }
+
+  public async getTigerGraphData (vertexType: Array<string>, edgeType: Array<string>): Promise<{ nodes: N[]; links: L[] }> {
+    return fetch(`${this.host}:14240/gsqlserver/interpreted_query`, {
+      method: 'POST',
+      body: `INTERPRET QUERY () FOR GRAPH ${this.graphname} {
+          ListAccum<EDGE> @@edges;
+          Seed = {${vertexType.join('.*, ')}.*};
+          Res = SELECT d FROM Seed:d - ((${edgeType.join(' | ')}):e) -> :t
+                  ACCUM @@edges += e;
+          PRINT Seed;
+          PRINT @@edges AS edges;
+        }`,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.username}:${this.password}`)}`,
+      },
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error! status: ${response.status}`)
+      }
+
+      return response.json()
+    }).then(data => {
+      const links: L[] = []
+      const nodes: N[] = []
+
+      if (data.error) {
+        throw new Error(`Error! status: ${data.message}`)
+      }
+
+      const vertices = data.results[0].Seed
+      const edges = data.results[1].edges
+      for (const vertex in vertices) {
+        nodes.push({
+          id: `${vertices[vertex].v_type}_${vertices[vertex].v_id}`,
+          v_id: `${vertices[vertex].v_id}`,
+          v_type: `${vertices[vertex].v_type}`,
+          ...(vertices[vertex].attributes),
+
+        })
+      }
+      for (const edge in edges) {
+        links.push({
+          source: `${edges[edge].from_type}_${edges[edge].from_id}`,
+          target: `${edges[edge].to_type}_${edges[edge].to_id}`,
+          ...(edges[edge].attributes),
+        })
+      }
+
+      return { nodes, links }
+    })
+  }
+
+  public async runInterpretedQuery (interpretedQuery: string): Promise<{ nodes: N[]; links: L[] }> {
+    return fetch(`${this.host}:14240/gsqlserver/interpreted_query`, {
+      method: 'POST',
+      body: interpretedQuery,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.username}:${this.password}`)}`,
+      },
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error! status: ${response.status}`)
+      }
+
+      return response.json()
+    }).then(data => {
+      const links: L[] = []
+      const nodes: N[] = []
+
+      if (data.error) {
+        throw new Error(`Error! status: ${data.message}`)
+      }
+
+      data = data.results
+
+      for (const res in data) {
```

Comment:
@GenericP3rson What if we use a more modern "for ... of" loop instead of "for ... in"?